bin/cobbler-settings

```diff
@@ -114,15 +116,40 @@ def modify(args: argparse.Namespace) -> int:
     schema = migrations.get_schema(migrations.get_installed_version())
     setting_obj = helper.Setting(args.key, args.value)
     # _schema used due to old version of python3-schema in Leap
-    key_type = schema._schema[setting_obj.key_name]
+    try:
+        key_type = schema._schema[setting_obj.key_name]
+    except KeyError:
+        key_found = False
+        for entry in schema._schema:
+            if isinstance(entry, Optional):
+                if entry._schema == setting_obj.key_name:
+                    key_type = schema._schema[entry]
+                    key_found = True
+                    break
+        if not key_found:
+            print("Requested key not found in the settings!")
+            return 1
+    if key_type not in (bool, int, float, str, dict, list):
+        if isinstance(key_type, list):
+            key_type = list
+        elif isinstance(key_type, dict):
+            key_type = dict
+        else:
+            print("Unsupported type!")
+    if isinstance(Optional, key_type):
+        key_type = schema._schema[setting_obj.key_name].name
```

Comment:
Wasn't this access failing in the test suite? Why is it working here :confused:

Reply:
Actually, I think that this code is never executed because of our first `try`/`except`. Block. I will test to see if I can remove the code safely.

Reply:
Looking at our tests it seems we can leave it removed.